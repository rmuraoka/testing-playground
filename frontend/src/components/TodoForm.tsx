import React, { useState } from 'react';
import { TextField, Button, Box } from '@mui/material';

interface TodoFormProps {
    onAddTodo: (todo: { title: string; description?: string }) => void;
}

const TodoForm: React.FC<TodoFormProps> = ({ onAddTodo }) => {
    const [title, setTitle] = useState<string>('');
    const [description, setDescription] = useState<string>('');

    const handleSubmit = () => {
        if (title) {
            onAddTodo({ title, description });
            setTitle(''); // フォームをクリア
            setDescription(''); // フォームをクリア
        }
    };

    return (
        <Box display="flex" flexDirection="column" mb={2}>
            <TextField
                data-testid="title"
                label="Title"
                variant="outlined"
                value={title}
                onChange={(e) => setTitle(e.target.value)}
                margin="normal"
            />
            <TextField
                data-testid="description"
                label="Description"
                variant="outlined"
                value={description}
                onChange={(e) => setDescription(e.target.value)}
                margin="normal"
            />
            <Button
                data-testid="add-task-button"
                variant="contained"
                color="primary"
                onClick={handleSubmit}
            >
                Add Task
            </Button>
        </Box>
    );
};

export default TodoForm;
