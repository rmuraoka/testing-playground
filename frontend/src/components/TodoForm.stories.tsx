import React from 'react';
import {Meta} from '@storybook/react';
import TodoForm from "./TodoForm";

export default {
    title: 'TodoFrom',
    component: TodoForm,
} as Meta;

export const Default = () =>
    <TodoForm
        onAddTodo={(todo) => console.log('Todo added:', todo)}
    />;
