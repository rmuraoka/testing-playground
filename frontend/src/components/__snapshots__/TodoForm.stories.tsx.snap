// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoFrom Default smoke-test 1`] = `
<div class="MuiBox-root css-1rplq84">
  <div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1fw0kwd-MuiFormControl-root-MuiTextField-root"
       data-testid="title"
  >
    <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
           data-shrink="false"
           for=":r0:"
           id=":r0:-label"
    >
      Title
    </label>
    <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root">
      <input aria-invalid="false"
             id=":r0:"
             type="text"
             class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
             value
      >
      <fieldset aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
      >
        <legend class="css-81qg8w">
          <span>
            Title
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1fw0kwd-MuiFormControl-root-MuiTextField-root"
       data-testid="description"
  >
    <label class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
           data-shrink="false"
           for=":r1:"
           id=":r1:-label"
    >
      Description
    </label>
    <div class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root">
      <input aria-invalid="false"
             id=":r1:"
             type="text"
             class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
             value
      >
      <fieldset aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
      >
        <legend class="css-81qg8w">
          <span>
            Description
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-74d805-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
          id=":r2:"
          data-testid="add-task-button"
  >
    Add Task
  </button>
</div>
`;
